src/mo_types/type.ml

```diff
@@ -1399,6 +1414,18 @@ let glb t1 t2 = let glbs = ref M.empty in combine glbs (ref M.empty) glbs t1 t2
 
 module Env = Env.Make(String)
 
+(* Stable function support *)
+
+type captured_variable = {
+  stable_name: string; (* positional id for parameters, named id for locals *)
+  variable_type: typ;
+}
+
+type stable_closure = {
+  function_path: string list; (* fully qualified function name *)
+  captured_variables: captured_variable Env.t; (* captured mutable variables *)
```

Comment:
```suggestion
  captured_variables: captured_variable Env.t; (* captured mutable variables *)
```
Just the mutable ones?

Reply:
It actually captures only mutable variables that then implicitly become stable. Immutable ones are not part of the closures, directly referenced or copied.